fenwick/src/main/java/org/opencadc/fenwick/StorageSiteSync.java

```diff
@@ -0,0 +1,204 @@
+ * org.opencadc.fenwick.trackSiteLocations property in the fenwick.properties file.
+ */
+public class StorageSiteSync {
+    private static final Logger log = Logger.getLogger(StorageSiteSync.class);
+
+    private static final String STORAGE_SITE_QUERY = "SELECT name, id, resourceid FROM inventory.storagesite";
+
+    private final TapClient<StorageSite> tapClient;
+    private final StorageSiteDAO storageSiteDAO;
+
+    /**
+     * Constructor that accepts a TapClient to query Storage Sites to sync.  Useful for testing.
+     *
+     * @param tapClient The TapClient to use.
+     */
+    public StorageSiteSync(final TapClient<StorageSite> tapClient, final StorageSiteDAO storageSiteDAO) {
+        this.tapClient = tapClient;
+        this.storageSiteDAO = storageSiteDAO;
+    }
+
+    /**
+     * Query for a Storage Site from a remote TAP (Luskan) service.  This method is expected to return a single
+     * StorageSite instance.  The result will typically be used to set to an Artifact's site locations during metadata
+     * synchronization.
+     *
+     * @return StorageSite  There should be a single StorageSite found.
+     * @throws AccessControlException permission denied
+     * @throws NotAuthenticatedException authentication attempt failed or rejected
+     * @throws ByteLimitExceededException input or output limit exceeded
+     * @throws IllegalArgumentException null method arguments or invalid query
+     * @throws ResourceNotFoundException remote resource not found
+     * @throws TransientException temporary failure of TAP service: same call could work in future
+     * @throws IOException failure to send or read data stream
+     * @throws InterruptedException thread interrupted
+     */
+    public StorageSite doit() throws AccessControlException, ResourceNotFoundException,
+                                     ByteLimitExceededException, NotAuthenticatedException,
+                                     IllegalArgumentException, TransientException, IOException,
+                                     InterruptedException {
+        try (final ResourceIterator<StorageSite> storageSiteIterator = queryStorageSites()) {
+            if (storageSiteIterator.hasNext()) {
+                final StorageSite storageSite = storageSiteIterator.next();
+                if (storageSiteIterator.hasNext()) {
+                    throw new IllegalStateException("More than one Storage Site found.");
+                } else {
+                    putStorageSite(storageSite);
+                    return storageSite;
+                }
+            } else {
+                throw new IllegalStateException("No storage sites available to sync.");
+            }
+        }
+    }
+
+    /**
+     * Store the StorageSite in the local inventory database.
+     * @param storageSite   The StorageSite to store.
+     */
+    void putStorageSite(final StorageSite storageSite) {
```

Comment:
This is a lot of code for a 1-line autocommit transaction. You only need explicit transaction handling if you are doing more than one change via DAO classes and need it to be atomic.

Please just inline the single line up at line~144.